KachnaOnline.Business.Data/Repositories/EventsRepository.cs

```diff
@@ -0,0 +1,52 @@
+// EventsRepository.cs
+// Author: David Chocholatý
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using KachnaOnline.Business.Data.Repositories.Abstractions;
+using KachnaOnline.Data;
+using KachnaOnline.Data.Entities.Events;
+using Microsoft.EntityFrameworkCore;
+
+namespace KachnaOnline.Business.Data.Repositories
+{
+    public class EventsRepository : GenericRepository<Event, int>, IEventsRepository
+    {
+        public EventsRepository(AppDbContext dbContext) : base(dbContext)
+        {
+        }
+
+        public IAsyncEnumerable<Event> GetCurrent(DateTime? atTime = null)
+        {
+            var actualAtTime = atTime ?? DateTime.Now;
+
+            return Set
+                .Where(e => e.From <= actualAtTime && e.To >= actualAtTime)
+                .AsAsyncEnumerable();
+        }
+
+        public IAsyncEnumerable<Event> GetNearest(DateTime? after = null)
+        {
+            var afterDate = after ?? DateTime.Now;
+
+            var eventEntity = Set.Where(e => e.From > afterDate).OrderBy(e => e.From).FirstOrDefaultAsync();
```

Comment:
Why isn't this already returned after the `OrderBy` as `AsAsyncEnumerable()`?

Reply:
Because the project will not compile and/or will throw runtime exceptions that should not be there. This is the only solution I was able to come up with to return async enumerable after a group by method.